### PR TITLE
6 add mechanism to read and globally save options

### PR DESCRIPTION
**Pull request by SirTalksalot75**
added mechanism to save and read options globally

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -50,6 +84,18 @@ void setup(void) {
 
 int main(int argc, char *argv[]) {
 
+   // Iteration through arguments
+    for (int i = 1; i < argc; i++) {
+        // Check if the argument starts with "--"
+        if (argv[i][0] == '-' && argv[i][1] == '-') {
+            // Extract option name
+            char option[10];
```
Comment (flixm16) [inline src/main.c:92]: what if my argv[i] is not 12 symbols long?

File: src/main.c [hunk 2]
```diff
@@ -7,12 +7,47 @@
 
 #define LOG_LEVEL LOG_LEVEL_DEBUG
 
+extern FILE *yyin;
+
+// Global array to store options
+char options[5][10];
```
Comment (flixm16) [inline src/main.c:13]: such a small limitation is bad. Limitations are bad. what if i want to have 20 Options? what is with 30? 2000? i dont want to change it all the time

File: src/main.c [hunk 3]
```diff
@@ -50,6 +84,18 @@ void setup(void) {
 
 int main(int argc, char *argv[]) {
 
+   // Iteration through arguments
+    for (int i = 1; i < argc; i++) {
+        // Check if the argument starts with "--"
+        if (argv[i][0] == '-' && argv[i][1] == '-') {
+            // Extract option name
+            char option[10];
+            strcpy(option, argv[i] + 2);
```
Comment (flixm16) [inline src/main.c:93]: why do you copy it? argv and argc has the same informations. save them global and use these.

File: src/main.c [hunk 4]
```diff
@@ -50,6 +84,18 @@ void setup(void) {
...
+
+            // Add option to the global array
+            add_option(option);
```
Comment (flixm16) [inline src/main.c:96]: why do you copy argv in options? is it possible to put argv in add_option?

File: src/main.c [hunk 5]
```diff
@@ -7,12 +7,47 @@
...
 void notify_exit(void) { DEBUG("Exiting gemstone..."); }
 
+
+/**
+ * @brief add option to global option array
+ *
+ */
+
+void add_option(const char* option) {
+    if (num_options < 5 ) {
```
Comment (flixm16) [inline src/main.c:29]: i also have to change the number of options here?

**Review by flixm16 — CHANGES_REQUESTED** [review]

**Comment by Servostar** [comment]
# Implementation proposal
(Based on a spontaneous idea)

May you please consider the following:
Instead of copying the arguments to a static buffer we can keep a copy of the `argc` value and `argv` pointer globally.
Since both are valid the entire duration of the app, we can search through them directly (instead of the static buffer).

```c
// store argc and arg from main globally
int argc;
char** argv;

size_t check_option(const char* name) {
    // directly search through argv 
    for (int i = 0; i < argc; i++) {
        if (strcmp(name, argv[i] + 2) == 0) {
            return 1;
        }
    }
    return 0;
}
```

**Comment by flixm16** [comment]
please resolve comments

File: src/main.c (hunk 1 above)
Comment (SirTalksalot75) [reply src/main.c:92]: the max size is 10, still works, its only the reservation that is 10.

File: src/main.c (hunk 2 above)
Comment (SirTalksalot75) [reply src/main.c:13]: ye, but its about the implementation of the function, the changes on the options can be handeled when needed.

File: src/main.c (hunk 5 above)
Comment (SirTalksalot75) [reply src/main.c:29]: ye, basically

File: src/main.c (hunk 2 above)
Comment (flixm16) [reply src/main.c:13]: why do i even need a restriction? make it variable

File: src/main.c (hunk 3 above)
Comment (flixm16) [reply src/main.c:93]: strcpy results in undefined behavior when the destination is smaller than the source. you should add a check befor using something like that

**Review by flixm16 — CHANGES_REQUESTED** [review]

**Comment by Servostar** [comment]
Development switched to @Servostar

**Pull request closed without merging**